src/machines/modelingMachine.ts

```diff
@@ -793,16 +793,21 @@ export const modelingMachine = createMachine(
           if (Object.keys(sceneEntitiesManager.activeSegments).length > 0) {
             await sceneEntitiesManager.tearDownSketch({ removeAxis: false })
           }
+          sceneInfra.resetMouseListeners()
           await sceneEntitiesManager.setupSketch({
             sketchPathToNode: sketchDetails?.sketchPathToNode || [],
             forward: sketchDetails.zAxis,
             up: sketchDetails.yAxis,
             position: sketchDetails.origin,
             maybeModdedAst: kclManager.ast,
           })
-          sceneEntitiesManager.setupSketchIdleCallbacks(
-            sketchDetails?.sketchPathToNode || []
-          )
+          sceneInfra.resetMouseListeners()
```

Comment:
I think you might have mentioned this in one of your videos so I'll go back and watch, but why do you have to call resetMouseListeners again here?

Reply:
Really just a safety thing, `setupSketchIdleCallbacks` doesn't set every single type of call back, like it uses `onDrag`, but maybe on `onLeave`, and so there's a chance an stale `onLeave` callback could still be running and causing weird behaviour.

But maybe I should change `sceneInfra.setCallbacks(` to work such that any callbacks that don't get set are still wiped clean. That's probably a safer api.

Reply:
https://github.com/KittyCAD/modeling-app/issues/1993